### ISSUE TITLE
[Feature Request] support for decimal 1.9.x

### ISSUE THREAD

**Issue by ananthakumaran** [ahamez/protox#89]
The library currently depends on decimal 2.0 which makes it hard to upgrade to the latest version since there are other dependencies that are still on 1.9.x. Other popular libraries like [ecto](https://github.com/elixir-ecto/ecto/blob/master/mix.exs#L36) support both versions. Since the usage of decimal is very low, it should be easier to support both. I can open a PR if you are ok with adding support.

**Comment by ahamez** [comment]
Thank you for this report, I didn't know it was possible to specify deps with `{:decimal, "~> 1.9 or ~> 2.0"}`. I think it's perfectly reasonable to relax this constraint.

However, I tried to compile protox with Decimal 1.9, but [it uses](https://github.com/ahamez/protox/blob/4c63788466a7b9abf12b0151f344324003f2c20d/lib/protox/json_decode.ex#L166) the [`integer?`](https://hexdocs.pm/decimal/2.0.0/Decimal.html#integer?/1) function which is only present in Decimal 2.0.

I guess it means we have to backport this function. What do you think?

**Comment by ananthakumaran** [comment]
```diff
diff --git a/lib/google/protobuf/duration.ex b/lib/google/protobuf/duration.ex
index 4815007..fbe70a1 100644
--- a/lib/google/protobuf/duration.ex
+++ b/lib/google/protobuf/duration.ex
@@ -52,8 +52,15 @@ defimpl Protox.JsonMessageDecoder, for: Google.Protobuf.Duration do
   # -- Private
 
   defp get_decimal(json) do
-    case Decimal.parse(json) do
-      {dec, "s"} -> dec
+    json_decimal =
+      case String.split_at(json, -1) do
+        {json_decimal, "s"} -> json_decimal
+        _ -> raise Protox.JsonDecodingError.new("invalid duration format")
+      end
+
+    case Decimal.parse(json_decimal) do
+      {:ok, dec} -> dec
+      {dec, ""} -> dec
       {_dec, _suffix} -> raise Protox.JsonDecodingError.new("invalid duration format")
       :error -> raise Protox.JsonDecodingError.new("invalid duration format")
     end
diff --git a/lib/protox/json_decode.ex b/lib/protox/json_decode.ex
index f1d0434..b9d1a43 100644
--- a/lib/protox/json_decode.ex
+++ b/lib/protox/json_decode.ex
@@ -163,7 +163,7 @@ defmodule Protox.JsonDecode do
   def decode_value(json_value, type) when is_protobuf_integer(type) and is_float(json_value) do
     decimal_float = Decimal.from_float(json_value)
 
-    if Decimal.integer?(decimal_float) do
+    if Decimal.equal?(Decimal.round(decimal_float), decimal_float) do
       decode_value(Decimal.to_integer(decimal_float), type)
     else
       raise JsonDecodingError.new(
```

I just tested the above patch with both 1.9 and 2.0 and it passes all the tests. Using round seems simpler compared to backporting that function, we anyhow deal with erlang float which can't represent inf or any other types of non fraction, round should work in our case. The parse function has also some difference which needs to be handled as well.

**Comment by ahamez** [comment]
Thank you! Could you please open a PR with this patch? I could apply it myself, but I prefer your work being credited to you ;-)

**Comment by ahamez** [comment]
The fix is now released in [1.6.5](https://hexdocs.pm/protox/1.6.5/readme.html).
